### DISCUSSION TITLE
Rotate github personal access token

### DISCUSSION THREAD

**Discussion by rlankfo** [fluxcd/flux2#2161] [Q&A] [answered]
What's the best way to rotate an expired github personal access token?

**Comment by stefanprodan** [comment, marked as answer]
To rotate the bootstrap key be it a token or a deploy key:
- delete the auth secret from the cluster with `kubectl -n flux-system delete secret flux-system`
- rerun `flux bootstrap` with the same args as before
- flux will generate a new secret and will update the deploy key if you’re using SSH deploy keys

**Reply by rlankfo** [reply]
thanks 🙏

**Reply by maksymlumen** [reply]
But how I can tell which bootstrap command was used back then?

**Reply by Fyko** [reply]
I'd agree with @maksymlumen, that's a little frightning. Is there an alternative?

Edit: ended up doing `EDITOR="code-insiders --wait" kubectl -n flux-system edit secret ssh-credentials`

**Reply by makkes** [reply]
The PAT isn't stored in the cluster. Rather the SSH deploy key generated by Flux is tied to that token. The SSH deploy key doesn't expire so if you want to run the bootstrap command again (which you should do regularly to upgrade Flux) the only thing to do is to regenerate the PAT and run `flux bootstrap` again. Nothing to be frightened about there.

**Reply by ha185101** [reply]
Thanks a lot ! It worked for me.
